### PR TITLE
Fix parentheses offence for numeric arguments with an operator

### PR DESCRIPTION
**Pull request by gsamokovarov**
In the `Style/MethodCallWithArgsParentheses` cop `omit_parentheses`
style, we register an offense for cases like `foo -1`  or `foo +1`. This
is a potential Ruby warning and an offense from
`Lint/AmbiguousOperator`.

### CODE REVIEW

File: lib/rubocop/cop/style/method_call_with_args_parentheses.rb [hunk 1]
```diff
@@ -371,6 +373,10 @@ def regexp_slash_literal?(node)
           node.regexp_type? && node.loc.begin.source == '/'
         end
 
+        def numeric_with_operator?(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/method_call_with_args_parentheses.rb]: I think this should be a node extension.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Changes
 
+* [#7107](https://github.com/rubocop-hq/rubocop/issues/7107): Fix parentheses offence for numeric arguments with an operator in `Style/MethodCallWithArgsParentheses`. ([@gsamokovarov][])
```
Comment (bbatsov) [inline CHANGELOG.md]: I think this should be under "Bugs fixed".

Comment (gsamokovarov) [reply CHANGELOG.md]: Oh, of course!

**Comment by bbatsov** [comment]
@gsamokovarov Looking at the second commit message I think you didn't squash your commits properly. ;-)

**Comment by gsamokovarov** [comment]
Oh, noes! My recent `git commit` scissor changes. Let me try again.

File: lib/rubocop/cop/style/method_call_with_args_parentheses.rb (hunk 1 above)
Comment (gsamokovarov) [reply lib/rubocop/cop/style/method_call_with_args_parentheses.rb]: I have extracted `NumericNodes#sign?` are you okay with it?

**Comment by bbatsov** [comment]
Thanks!
